src/AutoRest.CSharp/Common/Input/InputTypes/InputParameter.cs

```diff
@@ -45,7 +45,75 @@ internal record InputParameter(
         SkipUrlEncoding: false,
         Explode: false,
         ArraySerializationDelimiter: null,
-        HeaderCollectionPrefix: null,
-        SerializationFormat: SerializationFormat.Default)
+        HeaderCollectionPrefix: null)
     { }
+
+    internal InputParameter(
+        string name,
+        string nameInRequest,
+        string? description,
+        InputType type,
+        RequestLocation location,
+        InputConstant? defaultValue,
+        VirtualParameter? virtualParameter,
+        InputParameter? groupedBy,
+        InputOperationParameterKind kind,
+        bool isRequired,
+        bool isApiVersion,
+        bool isResourceParameter,
+        bool isContentType,
+        bool isEndpoint,
+        bool skipUrlEncoding,
+        bool explode,
+        string? arraySerializationDelimiter,
+        string? headerCollectionPrefix,
+        SerializationFormat serializationFormat) : this(
+            name,
+            nameInRequest,
+            description,
+            type,
+            location,
+            defaultValue,
+            virtualParameter,
+            groupedBy,
+            kind,
+            isRequired,
+            isApiVersion,
+            isResourceParameter,
+            isContentType,
+            isEndpoint,
+            skipUrlEncoding,
+            explode,
+            arraySerializationDelimiter,
+            headerCollectionPrefix)
+    {
+        _serializationFormat = serializationFormat;
```

Comment:
Shall we set the  _serializationFormat here? Since we will re-calculate it when use the property, we will not need to initialize it.

This internal constructor is not needed. When convert an inputParameter from modelerFour, we just leave the `_serializationFormat` null, and re-calculate it when use.

Reply:
Ideally, we should have everything working in the same way, which is: calculate the serialization format when we first access it based on its type.
The reason we have to have this is that in the code model converter, it is calculating the serialization format in the converter and passed it in. I do not want to change behavior there therefore I introduced this to make sure at least on the DPG side everything is working fine (nothing changed in the generated code proves it).
I do not want to dig into that part too much because this is a dependency for the sample refactor.

Reply:
I have a try and updated the code. Now code model converter no longer needs to pass in the serialization format calculated elsewhere. Now everything is calculated in `InputParameter` and nothing changed.
For more details if you are interested, you could check above hidden comment.